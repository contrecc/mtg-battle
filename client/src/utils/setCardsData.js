export default function setCardsData(json) {
  return function(state, props) {
    return {
      card1: {
        image_uris: {
          art_crop: json[0].image_uris.art_crop,
          normal: json[0].image_uris.normal,
          current: json[0].image_uris.art_crop
        },
        name: json[0].name,
        power: Number(json[0].power),
        toughness: Number(json[0].toughness)
      },
      card2: {
        image_uris: {
          art_crop: json[1].image_uris.art_crop,
          normal: json[1].image_uris.normal,
          current: json[1].image_uris.art_crop
        },
        name: json[1].name,
        power: Number(json[1].power),
        toughness: Number(json[1].toughness)
      }
    };
  };
}
