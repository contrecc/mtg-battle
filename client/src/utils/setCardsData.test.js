import setCardsData from './setCardsData';

describe('setCardsData', () => {
  const jsonData = [
    {
      image_uris: {
        art_crop:
          'https://img.scryfall.com/cards/art_crop/en/ima/193.jpg?1517813031',
        current:
          'https://img.scryfall.com/cards/art_crop/en/ima/193.jpg?1517813031',
        normal:
          'https://img.scryfall.com/cards/normal/en/ima/193.jpg?1517813031'
      },
      name: 'Bladewing the Risen',
      power: 4,
      toughness: 4
    },
    {
      image_uris: {
        art_crop:
          'https://img.scryfall.com/cards/art_crop/en/rix/172.jpg?1524752439',
        current:
          'https://img.scryfall.com/cards/art_crop/en/rix/172.jpg?1524752439',
        normal:
          'https://img.scryfall.com/cards/normal/en/rix/172.jpg?1524752439'
      },
      name: 'Storm Fleet Sprinter',
      power: 2,
      toughness: 2
    }
  ];

  it('should return a function', () => {
    expect(setCardsData(jsonData)).toMatchSnapshot();
  });

  it('should return a function that then updates the state', () => {
    const returnedFunction = setCardsData(jsonData);
    expect(returnedFunction({}, {})).toMatchSnapshot();
  });
});
