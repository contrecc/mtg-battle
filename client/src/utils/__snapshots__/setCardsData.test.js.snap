// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setCardsData should return a function 1`] = `[Function]`;

exports[`setCardsData should return a function that then updates the state 1`] = `
Object {
  "card1": Object {
    "image_uris": Object {
      "art_crop": "https://img.scryfall.com/cards/art_crop/en/ima/193.jpg?1517813031",
      "current": "https://img.scryfall.com/cards/art_crop/en/ima/193.jpg?1517813031",
      "normal": "https://img.scryfall.com/cards/normal/en/ima/193.jpg?1517813031",
    },
    "name": "Bladewing the Risen",
    "power": 4,
    "toughness": 4,
  },
  "card2": Object {
    "image_uris": Object {
      "art_crop": "https://img.scryfall.com/cards/art_crop/en/rix/172.jpg?1524752439",
      "current": "https://img.scryfall.com/cards/art_crop/en/rix/172.jpg?1524752439",
      "normal": "https://img.scryfall.com/cards/normal/en/rix/172.jpg?1524752439",
    },
    "name": "Storm Fleet Sprinter",
    "power": 2,
    "toughness": 2,
  },
}
`;
